src/sdbal/Vester.sol

```diff
@@ -0,0 +1,223 @@
+     function claim(uint256 _nonce) external onlyBeneficiary whenNotPaused {
+         VestingPosition storage vestingPosition = vestingPositions[_nonce];
+         if (vestingPosition.claimed) {
+             revert AlreadyClaimed();
+         }
+         if (block.timestamp < vestingPosition.vestingEnds) {
+             revert NotVestedYet();
+         }
+         vestingPosition.claimed = true;
+
+         // Claim BAL rewards
+         ILiquidityGauge(address(SD_BAL_GAUGE)).claim_rewards(address(this), msg.sender);
+
+         // Transfer staked BAL to beneficiary
+         SD_BAL_GAUGE.safeTransfer(msg.sender, vestingPosition.amount);
+
+         emit Claimed(_nonce, vestingPosition.amount);
+     }
+
+     /// @notice Deposit logic but with default vesting period
+     /// @param _amount Amount of tokens to deposit
+     function deposit(uint256 _amount) external onlyMaxisOps whenNotPaused {
+         _deposit(_amount, DEFAULT_VESTING_PERIOD);
+     }
+
+     /// @notice Deposit logic
+     /// @param _amount Amount of tokens to deposit
+     /// @param _vestingPeriod Vesting period in seconds
+     function deposit(uint256 _amount, uint256 _vestingPeriod) external onlyMaxisOps whenNotPaused {
+         _deposit(_amount, _vestingPeriod);
+     }
+
+     /// @notice Maxis msig should be able to sweep any ERC20 tokens except staked aura bal
+     /// @param _token Address of the token to sweep
+     /// @param _amount Amount of tokens to sweep
+     /// @param _to Address to send the tokens to
+     function sweep(address _token, uint256 _amount, address _to) external onlyMaxisOps {
+         if (_token == address(SD_BAL_GAUGE)) {
+             revert ProtectedToken();
+         }
+         ERC20(_token).safeTransfer(_to, _amount);
+         emit Sweep(_token, _amount, _to);
+     }
+
+     /// @notice Ragequit all in case of emergency
+     /// @dev This function is only callable by the DAO multisig
+     /// @param _to Address to send all BAL BAL and BAL to
+     function ragequit(address _to) external onlyDaoMsig {
+         // Claim rewards and transfer BAL to beneficiary
+         ILiquidityGauge(address(SD_BAL_GAUGE)).claim_rewards(address(this), _to);
+
+         // Transfer staked BAL BAL to beneficiary
+         SD_BAL_GAUGE.safeTransfer(_to, SD_BAL_GAUGE.balanceOf(address(this)));
+
+         // Transfer BAL to beneficiary.
+         BAL.safeTransfer(_to, BAL.balanceOf(address(this)));
+
+         // Transfer USDC to beneficiary.
+         USDC.safeTransfer(_to, USDC.balanceOf(address(this)));
+
+         // Pause and render the contract useless
+         _pause();
+         emit Ragequit(_to);
+     }
+
+     /// @notice Function to claim aura rewards from staked auraBAL
+     /// @notice Can be still claimed even if the contract is paused
+     function claimRewards() external onlyBeneficiary {
+         ILiquidityGauge(address(SD_BAL_GAUGE)).claim_rewards(address(this), beneficiary);
+     }
+
```

Comment:
A small amount of docs here explaining what these rewards are and where to find proofs would be very helpful.